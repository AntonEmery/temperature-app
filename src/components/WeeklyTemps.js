import React, { Component } from 'react';
import { BarChart } from 'react-easy-chart';


class WeeklyTemps extends Component {
  state = {  };

  render() {

    const { weeklyTemps } = this.props;
    console.log(weeklyTemps.reduce((a, b) => a + b.temp, 0)/weeklyTemps.length)


    return (
      // wait until array populated to render graph
     <BarChart
          axes
          colorBars
          width={450}
          height={400}
          data={[

          ]}
        />
    );
  }
}

export default WeeklyTemps;