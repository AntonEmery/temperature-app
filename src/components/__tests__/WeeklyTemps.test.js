import React from 'react';
import renderer from 'react-test-renderer';

import WeeklyTemps from '../WeeklyTemps';

describe('Weekly Temps Component', () => {
  it('Component should render given correct props', () => {
    let units = 'Fahrenheit'
    let weeklyTemps = [
      { tempC: 15, tempF: 80 },
      { tempC: 19, tempF: 75 },
      { tempC: 26, tempF: 60 },
      { tempC: 30, tempF: 65 },
      { tempC: 12, tempF: 80 },
      { tempC: 30, tempF: 55 },
    ]
    const tree = renderer.create(<WeeklyTemps units={units} weeklyTemps={weeklyTemps} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})