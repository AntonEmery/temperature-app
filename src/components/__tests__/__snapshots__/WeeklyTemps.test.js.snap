// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekly Temps Component Component should render given correct props 1`] = `
<div>
  <p>
    Average Weekly Temperature
  </p>
  <p>
    69
     degrees
  </p>
</div>
`;
